### PR TITLE
hwinfo: support exx32 family

### PR DESCRIPTION
**Pull request by sun681**
add hwinfo driver support for the
exx32 gecko family.

Signed-off-by: Sun Amar <sun681@gmail.com>

### CODE REVIEW

File: drivers/hwinfo/hwinfo_gecko.c [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <em_system.h>
+#include <drivers/hwinfo.h>
+#include <string.h>
+
+ssize_t z_impl_hwinfo_get_device_id(uint8_t *buffer, size_t length)
+{
+	uint64_t unique_id = SYSTEM_GetUnique();
```
Comment (alexanderwachter) [inline drivers/hwinfo/hwinfo_gecko.c:13]: I think you need to convert to big endian first.

**Review by alexanderwachter — CHANGES_REQUESTED** [review]

File: drivers/hwinfo/hwinfo_gecko.c (hunk 1 above)
Comment (sun681) [reply drivers/hwinfo/hwinfo_gecko.c]: yea I was not sure about thar but you are probably right. 
I've fixed it

**Review by alexanderwachter — APPROVED** [review]
LGTM. Thank you @sun681

**Comment by sun681** [comment]
> LGTM. Thank you @sun681

Thank you @alexanderwachter , you know more relevant reviewers to approve this pull request?

**Review by alexanderwachter — APPROVED** [review]

**Review by chrta — APPROVED** [review]
LGTM and successfully tested on efr32mg_sltb004a

**Review by pabigot — APPROVED** [review]
Looks good.

**Pull request merged**